br/pkg/utils/misc.go

```diff
@@ -157,3 +165,52 @@ func WithCleanUp(errOut *error, timeout time.Duration, fn func(context.Context)
 		log.Warn("Encountered but ignored error while cleaning up.", zap.Error(err))
 	}
 }
+
+func AllStackInfo() []byte {
+	res := make([]byte, 256*units.KiB)
+	for {
+		n := runtime.Stack(res, true)
+		if n < len(res) {
+			return res[:n]
+		}
+		res = make([]byte, len(res)*2)
+	}
+}
+
+var (
+	DumpGoroutineWhenExit atomic.Bool
+)
+
+func StartExitSingleListener(ctx context.Context) (context.Context, context.CancelFunc) {
+	cx, cancel := context.WithCancel(ctx)
+	sc := make(chan os.Signal, 1)
+	signal.Notify(sc,
```

Comment:
To confirm, will a normal init job exit print the go routines also?

Reply:
No, goroutine dump won't be printed once prepare done.